### PR TITLE
ROX-20175: Update all vulnerability requests to use image scope

### PR DESCRIPTION
**Pull request by md2119**
## Description

All vulnerability exception requests will be represented using `.scope.imageScope` fields starting 4.4. It simplifies the vulnerability request scope representation dramatically. The global scope exceptions will have `.scope.imageScope` as below:
```
{
  registry: ".*",
  remote: ".*",
  tag: ".*",
}
```
The global scope support for `Risk assessment workflow` (2nd generation exception management) was never announced. However, since the fields are available at APIs, we must ensure that we migrate any globally scoped vulnerability requests.
A notice for deprecation of `.scope.globalScope` is not required because `/v1/cve/requests` are deprecated and notified and `v2` APIs do not expose `storage.VulnerabilityRequest`.

**Note that this PR will not be merged until the 4.3 release is cut.**

## Checklist
- [ ] Investigated and inspected CI test results
- [X] Unit test and regression tests added
- ~Evaluated and added CHANGELOG entry if required~
- ~Determined and documented upgrade steps~
- ~Documented user-facing changes~

If any of these don't apply, please comment below.

## Testing Performed
Unit Test + Manually upgrade from 4.2.0 to PR build

4.2.0:
image

image

4.3.x-13-g89187eabdb:
image

### CODE REVIEW

**Comment by dashrews78** [comment]
First pass through, looks good.  I know of at least 2 other migrations in flight, so I will hold off approval until those migrations land and we have rebased off them.

File: migrator/migrations/m_194_to_m_195_vuln_request_global_scope/migration.go
Comment (dashrews78) [inline migrator/migrations/m_194_to_m_195_vuln_request_global_scope/migration.go:11]:
```suggestion
	startSeqNum = 194
```

**Review by dashrews78 — CHANGES_REQUESTED** [review]
startSeqNum in `migration.go`.
Other than that it looks good, but that one is important.

**Comment by md2119** [comment]
Did a manual upgrade from 4.2.0 to this PR build

**Comment by dashrews78** [comment]
FTR, I think https://github.com/stackrox/stackrox/pull/8346 busted the upgrade test.

**Comment by md2119** [comment]
> FTR, I think #8346 busted the upgrade test.


The eclipse continues

**Review by dashrews78 — APPROVED** [review]

**Pull request merged**
